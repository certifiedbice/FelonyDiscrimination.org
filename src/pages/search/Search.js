import React,{Component} from 'react'
import SearchForm from '../../components/searchform/SearchForm';
import SearchResultsList from '../../components/searchresultslist/SearchResultsList';
import './Search.css';

export default class Search extends Component{

	state={
		search_results:[],
		error:null,
	};

	storeSearchResults=res=>{this.setState({search_results:[...res]})}

	render(){
        return(
			<>
				<SearchForm storeSearchResults={this.storeSearchResults}/>
				<SearchResultsList search_results={this.state.search_results}/>
			</>
		);
	}
}