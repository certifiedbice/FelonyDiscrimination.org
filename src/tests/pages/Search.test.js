import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from '../../pages/search/Search';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<Search/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<Search/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});