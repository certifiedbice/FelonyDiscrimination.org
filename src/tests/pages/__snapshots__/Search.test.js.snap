// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section
    id="search-form-container"
  >
    <form
      aria-label="Search form"
      id="search-form"
      name="search-form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          <h2>
            Search
          </h2>
        </legend>
        <div
          className="search-form-element-container"
        >
          <label>
            Suggested search criteria: Name, City, State, Zipcode.
          </label>
          <input
            aria-labelledby="Search form email"
            className="search-form-element"
            id="search-form-input"
            name="text"
            placeholder="Search"
            required={true}
            type="text"
          />
        </div>
        <div
          className="search-form-error-container"
        />
        <div
          className="search-form-element-container"
        >
          <input
            aria-label="Search form submit"
            id="search-form-submit"
            required={true}
            type="submit"
            value="Submit"
          />
        </div>
      </fieldset>
    </form>
  </section>,
  <section
    id="search-results"
  />,
]
`;
